autogpt_platform/backend/backend/util/iffy.py

```diff
@@ -0,0 +1,150 @@
+from backend.util.settings import Settings
+from backend.data.db import prisma
+from backend.util.openrouter import moderate_content
+
+logger = logging.getLogger(__name__)
+settings = Settings()
+
+IFFY_API_KEY = settings.secrets.iffy_api_key
+IFFY_API_URL = settings.secrets.iffy_api_url
+
+class UserData(TypedDict):
+    clientId: str
+    email: Optional[str]
+    name: Optional[str]
+    username: Optional[str]
+
+async def send_to_iffy(user_id: str, block_content: Dict[str, Any]) -> Tuple[bool, str]:
+    """
+    Send block content to Iffy for content moderation.
+    Only used in cloud mode - local mode skips moderation entirely.
+    
+    Args:
+        user_id: The ID of the user executing the block
+        block_content: The content of the block to be moderated
+        
+    Returns:
+        Tuple[bool, str]: (is_safe, reason)
+        - is_safe: True if content was sent to Iffy successfully or passed OpenRouter moderation,
+                  False if both services failed or content was flagged
+        - reason: Description of the result or error
+    """
+    if settings.config.behave_as == "local":
+        logger.info("Content moderation skipped - running in local mode")
+        return True, "Moderation skipped - running in local mode"
+
+    # Validate Iffy API URL and key at the start
+    if not IFFY_API_URL or not IFFY_API_KEY:
+        logger.warning("Iffy API URL or key not configured, falling back to OpenRouter moderation")
+        is_safe, reason = await moderate_content(json.dumps(block_content.get('input_data', {}), indent=2))
+        if not is_safe:
+            logger.error(f"OpenRouter moderation failed after Iffy configuration issue: {reason}")
+        return is_safe, f"Iffy not configured. OpenRouter result: {reason}"
+
+    try:
+        # Validate URL format
+        if not IFFY_API_URL.startswith(('http://', 'https://')):
+            logger.error(f"Invalid Iffy API URL format: {IFFY_API_URL}")
+            return await moderate_content(json.dumps(block_content.get('input_data', {}), indent=2)), "Invalid Iffy API URL format"
+
+        headers = {
+            "Authorization": f"Bearer {IFFY_API_KEY}",
+            "Content-Type": "application/json"
+        }
+        
+        input_data = json.dumps(block_content.get('input_data', {}), indent=2)
+
+        # Get user details with proper connection handling
+        user_data: UserData = {
+            "clientId": user_id,
+            "email": None,
+            "name": None,
+            "username": None,
+        }
+
+        try:
+            if not prisma.is_connected():
+                await prisma.connect()
+            user = await get_user_by_id(user_id)
+            if user:
+                user_data.update({
+                    "email": user.email or user_data["email"],
+                    "name": user.name,
+                    "username": user.username if hasattr(user, 'username') else user.name,
+                })
```

Comment:
does this have to happen in the try catch or can it happen closer to where its used?